Makefile

```diff
@@ -63,7 +63,10 @@ tag:
 	git commit -m '$(TAG) release'
 	chag tag
 
-static: static-phpstan static-codestyle-check
+static: static-phpstan static-psalm static-codestyle-check
+
+static-psalm:
+	docker run --rm -it -v ${PWD}:/app -w /app vimeo/psalm-github-actions
```

Comment:
will produce different results to using the phar. can we have the makefile download the phar to /tmp or something?

Reply:
Yes, it might. But in this case it is a good thing, right? 

If the output is different then we need to be more defensive in our programming and that is never bad.

Reply:
From what I can see, this is 3 months old, so running a buggy local version would not be a good idea?

Reply:
Better to have no local version than not the same as the CI I think?

Reply:
Another option is to do this for both static analysis tools: https://github.com/vlucas/phpdotenv/blob/4.1/composer.json#L26, https://github.com/vlucas/phpdotenv/blob/4.1/vendor-bin/phpstan/composer.json.

Reply:
The vimeo psalm docker image is updated every 4 hours. 

https://hub.docker.com/r/vimeo/psalm-github-actions
https://github.com/psalm/psalm-github-actions/blob/master/.github/workflows/build.yml

I dont mind removing this from the Makefile all together. I dont feel like it is a good idea spending time on this.. I mean.. Power users have quick commands to run psalm anyways (like installing it globally) and others will have a Github review comment on the exact line with the issue.